www/js/controllers/customer_menu_controller.js

```diff
@@ -23,10 +23,52 @@ app.controller('CustomerMenuCtrl', function($scope, $state, $ionicLoading, UserS
   });
 
   $scope.openCharge = function() {
-    $state.go('app.transaction', {'transactionType': 'charge'});
+    var chargeFn = function() {
+      $state.go('app.transaction', {'transactionType': 'charge'});
+    };
+
+    if (CashierModeService.canCharge()) {
+      chargeFn();
+    } else {
+      executeAction(chargeFn);
+    }
   };
 
+
   $scope.openRefund = function() {
-    $state.go('app.transaction', {'transactionType': 'refund'});
+    //$state.go('app.transaction', {'transactionType': 'refund'});
+    var refundFn = function() {
+      $state.go('app.transaction', {'transactionType': 'refund'});
+    };
+
+    if (CashierModeService.canRefund()) {
+      refundFn();
+    } else {
+      executeAction(refundFn);
+    }
+  };
+
+
+  var executeAction = function(fn) {
+    NotificationService.showConfirm({
+      title: 'cashier_permission',
+      subTitle: "",
+      okText: "confirm",
+      cancelText: "cancel"
+    }, {}).then(function(res) {
+      if (res) {
+        return PermissionService.authorizeSeller()
+          .then(function(authResult) {
+            if (!authResult) {
+              NotificationService.showAlert({title: 'cashier_permission_rejected'});
+              return;
+            }
+            //$state.go('app.transaction', {'transactionType': 'charge'});
```

Comment:
This is the function that validates the seller to enable or not to Refund, to Charge, etc. It's the same for all preferences, except the function to execute, like, refund, charge. So, instead of repeat all code, i just pass a function with a closure that knows what to do.